pallets/subtensor/src/staking/stake_utils.rs

```diff
@@ -457,155 +457,148 @@ impl<T: Config> Pallet<T> {
-        });
-        // Step 5: Increase Alpha outstanding.
-        SubnetAlphaOut::<T>::mutate(netuid, |total| {
-            *total = total.saturating_add(alpha.saturating_to_num::<u64>());
-        });
-        // Step 6: Increase Tao reserves.
-        SubnetTAO::<T>::mutate(netuid, |total| {
-            *total = total.saturating_add(tao);
-        });
-        // Step 7: Increase Total Tao reserves.
-        TotalStake::<T>::mutate(|total| {
-            *total = total.saturating_add(tao);
-        });
-        // Step 8. Decrease Alpha reserves.
-        SubnetVolume::<T>::mutate(netuid, |total| {
-            *total = total.saturating_sub(tao);
-        });
-        // Step 9. Return the alpha received.
-        alpha.saturating_to_num::<u64>()
+            0
+        }
     }
 
     /// Swaps a subnet's Alpba token for TAO.
     ///
     /// Updates TaoIn, AlphaIn, and AlphaOut
     pub fn swap_alpha_for_tao(netuid: u16, alpha: u64) -> u64 {
-        // Step 1: Get the mechanism type for the subnet (0 for Stable, 1 for Dynamic)
-        let mechanism_id: u16 = SubnetMechanism::<T>::get(netuid);
-        // Step 2: Swap alpha and attain tao
-        let tao: I96F32 = if mechanism_id == 1 {
-            // Step 3.a.1: Dynamic mechanism calculations
-            let tao_reserves: I96F32 = I96F32::saturating_from_num(SubnetTAO::<T>::get(netuid));
-            let alpha_reserves: I96F32 =
-                I96F32::saturating_from_num(SubnetAlphaIn::<T>::get(netuid));
-            // Step 3.a.2: Compute constant product k = alpha * tao
-            let k: I96F32 = alpha_reserves.saturating_mul(tao_reserves);
-            // Step 3.a.3: Calculate alpha staked using the constant product formula
-            // tao_recieved = tao_reserves - (k / (alpha_reserves + new_tao))
-            tao_reserves.saturating_sub(
-                k.checked_div(alpha_reserves.saturating_add(I96F32::saturating_from_num(alpha)))
-                    .unwrap_or(I96F32::saturating_from_num(0)),
-            )
+        if let Some(tao) = Self::sim_swap_alpha_for_tao(netuid, alpha) {
+            // Step 4: Increase Alpha reserves.
+            SubnetAlphaIn::<T>::mutate(netuid, |total| {
+                *total = total.saturating_add(alpha);
+            });
+            // Step 5: Decrease Alpha outstanding.
+            SubnetAlphaOut::<T>::mutate(netuid, |total| {
+                *total = total.saturating_sub(alpha);
+            });
+            // Step 6: Decrease tao reserves.
+            SubnetTAO::<T>::mutate(netuid, |total| {
+                *total = total.saturating_sub(tao);
+            });
+            // Step 7: Reduce total TAO reserves.
+            TotalStake::<T>::mutate(|total| {
+                *total = total.saturating_sub(tao);
+            });
+            // Step 8. Decrease Alpha reserves.
+            SubnetVolume::<T>::mutate(netuid, |total| {
+                *total = total.saturating_sub(tao);
+            });
```

Comment:
Why are we decreasing the volume?
Shouldn't volume always be additive?

Looks like we've been subtracting from it in both places too....
```suggestion
            SubnetVolume::<T>::mutate(netuid, |total| {
                *total = total.saturating_add(tao);
            });
```